### PR TITLE
OCP4: api_server_no_adm_ctrl_plugins_disabled ignore PodSecurity

### PR DESCRIPTION
**Pull request by Vincent056**
We are going to ignore PodSecurity from the disabled admission control plugins list because it is validating only and it is disabled by default on 4.10 due to the transition from SCCs to PSs

Related Link: https://github.com/openshift/enhancements/pull/899/files

### CODE REVIEW

File: applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/rule.yml [hunk 1]
```diff
@@ -32,21 +35,23 @@ ocil_clause: 'No admission plugins are disabled'
 
 ocil: |-
     To verify that the list of disabled admission plugins is empty, run the following command:
-    <pre>$ oc -n openshift-kube-apiserver get configmap config -o json | jq -r '.data."config.yaml"' | jq '.apiServerArguments."disable-admission-plugins"'</pre>
+    <pre>$ <pre>$ oc -n openshift-kube-apiserver get mconfigmap config -o json | jq -r '{{{ jqfilter }}}'</pre>
```
Comment (rhmdnd) [inline applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/rule.yml:38]: It looks like we have an extra `<pre>$` here?

File: applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/tests/no_disabled.pass.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+yum install -y jq
+
+kube_apipath="/kubernetes-api-resources"
+
+mkdir -p "$kube_apipath/api/v1/namespaces/openshift-kube-apiserver/configmaps"
+
+config_apipath="/api/v1/namespaces/openshift-kube-apiserver/configmaps/config"
+
+# This file assumes that we have two routes outside the openshift/kube namespaces, and 
+# one route has rate limit enabled, and one route does not
```
Comment (rhmdnd) [inline applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/tests/no_disabled.pass.sh:14]: Is this comment valid for this change? Looks the same in each test proposed here, but we're not dealing with routes.

File: applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/tests/not_podsecurity.fail.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+yum install -y jq
+
+kube_apipath="/kubernetes-api-resources"
+
+mkdir -p "$kube_apipath/api/v1/namespaces/openshift-kube-apiserver/configmaps"
+
+config_apipath="/api/v1/namespaces/openshift-kube-apiserver/configmaps/config"
+
+# This file assumes that we have two routes outside the openshift/kube namespaces, and 
+# one route has rate limit enabled, and one route does not
```
Comment (rhmdnd) [inline applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/tests/not_podsecurity.fail.sh:14]: Similar comment here.

File: applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/tests/only_podsecurity.pass.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+yum install -y jq
+
+kube_apipath="/kubernetes-api-resources"
+
+mkdir -p "$kube_apipath/api/v1/namespaces/openshift-kube-apiserver/configmaps"
+
+config_apipath="/api/v1/namespaces/openshift-kube-apiserver/configmaps/config"
+
+# This file assumes that we have two routes outside the openshift/kube namespaces, and 
+# one route has rate limit enabled, and one route does not
```
Comment (rhmdnd) [inline applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/tests/only_podsecurity.pass.sh:14]: Similar comment here.

File: applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/tests/other_than_podsecurity.fail.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+yum install -y jq
+
+kube_apipath="/kubernetes-api-resources"
+
+mkdir -p "$kube_apipath/api/v1/namespaces/openshift-kube-apiserver/configmaps"
+
+config_apipath="/api/v1/namespaces/openshift-kube-apiserver/configmaps/config"
+
+# This file assumes that we have two routes outside the openshift/kube namespaces, and 
+# one route has rate limit enabled, and one route does not
```
Comment (rhmdnd) [inline applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/tests/other_than_podsecurity.fail.sh:14]: Similar comment here.

**Review by rhmdnd — COMMENTED** [review]
The initial pass looks OK to me, but I haven't tested it, yet.

Only minor suggestions on cleaning up the comments in some of the files.

File: applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/rule.yml (hunk 1 above)
Comment (Vincent056) [reply applications/openshift/api-server/api_server_no_adm_ctrl_plugins_disabled/rule.yml:38]: thanks for the review! Let me fix those!

**Comment by mrogers950** [comment]
Looks good.

**Comment by rhmdnd** [comment]
Looks good to me, my comments have been addressed.

File: applications/openshift/networking/default_ingress_ca_replaced/tests/ocp4/e2e-remediation.sh [hunk 6]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
-# You can safely patch this to a non-existent secret name, it's a no-op, but OK for testing.
-oc patch proxies.config cluster --type merge -p '{"spec":{"trustedCA":{"name":"testing"}}}'
+# You can safely patch this to a exist default secret name
```
Comment (rhmdnd) [inline applications/openshift/networking/default_ingress_ca_replaced/tests/ocp4/e2e-remediation.sh:3]: nit: maybe adding a quite note that a valid CA is required so this remediation doesn't cause subsequent failures.

Comment (Vincent056) [reply applications/openshift/networking/default_ingress_ca_replaced/tests/ocp4/e2e-remediation.sh]: that's a good idea, just added some comments to it

**Review by jhrozek — APPROVED** [review]
The code looks good and everyone's comments have been addressed.

**Pull request merged**